Feature/persist browsing data

crsct opened the pull request:
closes #73

crsct committed:
feature: persist browsing data by adding variables to query

crsct committed:
fix double calling of filter function upon init

crsct committed:
fix most double requests

crsct committed:
remove console.log statement

crsct committed:
fix remaining issue and use props passed by route

crsct committed:
Merge branch 'feature/persist-browsing-data' of github.com:ag-computational-bio/bakrep-web into feature/persist-browsing-data

lukasjelonek reviewed (changes requested):
I added some comments that may be worth discussiong/thinking about

crsct committed:
fix warning about type missmatch by parsing route query prop directly

crsct committed:
fix warning about type missmatch by parsing route query prop directly

crsct committed:
Merge branch 'feature/persist-browsing-data' of github.com:ag-computational-bio/bakrep-web into feature/persist-browsing-data

crsct committed:
fix error

crsct committed:
draft commit

crsct committed:
Merge branch 'main' of github.com:ag-computational-bio/bakrep-web into feature/persist-browsing-data

crsct committed:
add pagestate and min and max values by querying searchinfo

crsct committed:
fix conversations by removing nested min, max and rename updateQuery function to updateUrl

crsct committed:
fix pagination not working on first selection by adding parameters to the url immediately

crsct committed:
remove props from the router

crsct committed:
rename decodeQuery to parseFiltersFromRoute and reuse function populateVariables

crsct committed:
fix regression caused by wrong function call

crsct committed:
update getTupleRange to check if info exists and contains min and max values

crsct committed:
WIP commit

crsct committed:
added watch for individial query parameters and conditionalized uploadurl

crsct committed:
added offset to paramater comparison

crsct committed:
add first loop replace instead of push

crsct committed:
reset pagination on filter apply

crsct committed:
replace url only if a query was not set

crsct committed:
replace replace variable by using length of query object

lukasjelonek committed:
Rename FilterTuple to Range
Save filter values in an reactive object instead of single reactive variables
Replace `quality` with `completeness`
Add comments where it is not clear why something happens
Reduce parameters for route-query watcher

lukasjelonek committed:
Merge branch 'main' into feature/persist-browsing-data

lukasjelonek committed:
Reduce similar code blocks by introducing functions
Hopefully this improve readability

lukasjelonek committed:
Allow user to update the table size via the url
